projects/core/src/routing/store/reducers/router.reducer.spec.ts

```diff
@@ -199,4 +215,32 @@ describe('Router Reducer', () => {
       cmsRequired: false
     });
   });
+
+  describe('should set correct context for content pages', () => {
+    let context;
+
+    beforeEach(async () => {
+      store.subscribe(routerStore => {
+        context = routerStore.router.state.context;
+      });
+    });
+
+    it('for generic page', async () => {
+      await zone.run(() => router.navigateByUrl('/customCmsPage'));
+      expect(context).toEqual({
+        id: '/customCmsPage',
+        type: PageType.CONTENT_PAGE
+      });
+    });
+
+    it('for route defined with page label', async () => {
+      await zone.run(() => router.navigateByUrl('/cmsPage'));
+      expect(context).toEqual({ id: 'cmsPage', type: PageType.CONTENT_PAGE });
+    });
+
+    it('for route with cxCmsContext context', async () => {
+      await zone.run(() => router.navigateByUrl('/dynamically-created'));
+      expect(context).toEqual({ id: 'explicit', type: PageType.CONTENT_PAGE });
+    });
+  });
```

Comment:
It would be great to have unit tests for nested routes - because a lot of CustomSerializer's complex logic is addressing nested routes.

Reply:
👍